### ISSUE TITLE
Move ORCID functionality into core application

### ISSUE THREAD

**Issue by ewhanson** [pkp/pkp-lib#9771]
The ORCID functionality currently exists as part of the [OrcidProfilePlugin](https://github.com/pkp/orcidprofile). This issue is to track the work needed to move ORCID functionality entirely within the core application.

This includes:

- [x] OJS ORCID functionality moved
- [x] OPS ORCID functionality moved
- [x] Basic (public API) ORCID functionality added to OMP

**Comment by ewhanson** [comment]
Hey @bozana, here is the initial pass at this work. Could you have a look and let me know what you think before I clean it up/proceed with implementing this for OMP and OPS? Thanks!

PRs:

- `ojs`: pkp/ojs#4221
- `pkp-lib`: #9818
- `ui-library`:pkp/ui-library#339

**Comment by bozana** [comment]
Hi @ewhanson, I went through the code. All looks good. I haven't tested it. There are mostly smaller comments to consider and a few questions (eventually to think about). 
I think I had difficulties to understand classes/orcid/actions/AuthorizeUserData.php and classes/orcid/actions/VerifyAuthorWithOrcid.php -- not the code itself, but how ORCID is functioning there, what everything is how needed to be... I suppose I would need to read and dive deeper into ORCID documentation. However, if you find a way to make it easier for other developers to understand it, e.g. by providing a link to that ORCID API or documentation, it will be great, but no too much worries about it :-)
Thanks a lot, great work!

**Comment by bozana** [comment]
These changes needs to be consider once https://github.com/pkp/pkp-lib/issues/699 is merged: https://github.com/jyhein/orcidProfile/commit/a35e900f7b37f70b2aa1d74f5b32caa997c1f6d3.

**Comment by ewhanson** [comment]
Hey @bozana, this is finally ready for a review! Would you be able to have a look (complete PRs referenced above)? There are still a few things left to do, but I'd prefer to get these large set of changes merged and complete the rest as separate PRs (outlined below).

Still TODO:
- [ ] OpenAPI documentation for ORCID API endpoints
- [ ] `FieldOrcid` modal closing parent modal bug (in ui-library)
- [ ] Settle on pattern for submitting ORCID settings form with incomplete required field when disabling ORCID functionality
- [ ] Add additional integration tests

Thanks!

**Comment by jardakotesovec** [comment]
@ewhanson Thats great overview, thank you!

> The most common way for actual users will likely be via a job that's triggered at various points as a submission moves through the workflow process (e.g. on being accepted, on being published, etc.).

Can you explain this one in more detail? Does it mean that would receive email at some point (accepted/published) that would ask them to verify orcid?

**Comment by ewhanson** [comment]
Hey @jardakotesovec,

> Can you explain this one in more detail? Does it mean that would receive email at some point (accepted/published) that would ask them to verify orcid?

Yes, I actually just checked and I believe the "on publication" aspect needs to be reimplemented (I think it got lost in some refactoring), but the email is sent when the submission is accepted or skips review. You can see that code here: https://github.com/ewhanson/pkp-lib/blob/orcid/classes/observers/listeners/SendAuthorOrcidEmail.php

**Comment by ewhanson** [comment]
Merged!

**Comment by bozana** [comment]
@ewhanson, do we need to open a new issue for the further task(s) we still need or would like to do? E.g. OpenAPI documentation for ORCID API endpoints, consider the comments from the PKP-ORCID Working Group, what to do with old and not verified ORCID iDs in the DB (and consider import/export regarding this), double check/improve the language for some locale keys, e.g. orcid.verify.duplicateOrcid and orcid.verify.duplicateOrcidAuthor, could we provide a solution for a work to be pushed after it is published?... -- those are I know we had on the list, but I am not sure any more about their status, maybe some are already solved...
Thanks a lot!
